lib/src/drawConfigContext.dart

```diff
@@ -0,0 +1,282 @@
+/// Copyright (c) 2017, the Dart Reddit API Wrapper project authors.
+/// Please see the AUTHORS file for details. All rights reserved.
+/// Use of this source code is governed by a BSD-style license that
+/// can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:ini/ini.dart';
+import 'package:path/path.dart' as path;
+
+import './exceptions.dart';
+
+const String kFileName = 'draw.ini';
+
+const String kMacEnvVar = 'HOME';
+const String kLinuxEnvVar = 'XDG_CONFIG_HOME';
+const String kWindowsEnvVar = 'APPDATA';
+
+const String kShortUrl = 'short_url';
+const String kCheckForUpdates = 'check_for_updates';
+const String kKind = 'kind';
+const String kOptionalField = 'optional_field';
+const String kRequiredField = 'required_field';
+const String kComment = 'comment';
+const String kMessage = 'message';
+const String kRedditor = 'redditor';
+const String kSubmission = 'submission';
+const String kSubReddit = 'subreddit';
+const String kClientId = 'client_id ';
+const String kClientSecret = 'child_secret ';
+const String kHttpProxy = 'http_proxy ';
+const String kHttpsProxy = 'https_proxy ';
+const String kRedirectUri = 'redirect_uri ';
+const String kRefreshToken = 'refresh_token ';
+const String kPassword = 'password ';
+const String kUserAgent = 'user_agent ';
+const String kUsername = 'username ';
+const String kOauthUrl = 'oauth_url';
+const String kRedditUrl = 'reddit_url';
+
+const String kDefaultShortUrl = 'https://redd.it';
+
+final kNotSet = null;
+
+/// The [DrawConfigContext] class provides an iterface to store.
+/// Load the DRAW's configuration file [draw.ini].
+class DrawConfigContext {
+  static final Map<String, List<String>> fieldMap = {
+    kShortUrl: [kShortUrl],
+    kCheckForUpdates: [kCheckForUpdates],
+    kKind: [kComment, kMessage, kRedditor, kSubmission, kSubReddit],
+    kOptionalField: [
+      kClientId,
+      kClientSecret,
+      kHttpProxy,
+      kHttpsProxy,
+      kRedirectUri,
+      kRefreshToken,
+      kPassword,
+      kUserAgent,
+      kUsername,
+    ],
+    kRequiredField: [kOauthUrl, kRedditUrl]
+  };
+
+  /// Path to Local, User, Global Config Files, with matching precedence.
+  Uri _localConfigPath;
+  Uri _userConfigPath;
+  Uri _globalConfigPath;
+
+  Config _customConfig;
+
+  Map<String, String> custom;
+
+  String _shortURL;
+  String _primarySiteName;
+
+  bool checkForUpdates;
+  String userAgent;
+
+  String _redirectUri;
+  String clientId;
```

Comment:
These shouldn't be modifiable outside of this class. Either make them `final` and initialize them inside the constructor or make them private fields and create getters for them.